### PR TITLE
[dune] [dbg] Add support for coqtop in dune-dbg

### PR DESCRIPTION
**Pull request by ejgallego**
We also workaround problem #11405 , however, this should be reverted
once the problem is fixed in OCaml upstream.

### CODE REVIEW

File: dev/doc/build-system.dune.md [hunk 1]
```diff
@@ -108,24 +108,38 @@ automatically.
...
-[it will then correctly compute the deps and rebuild if you call the
-script again] This will be fixed in the future.
+Unfortunately, dependency handling here is not fully refined, so you
+need to build enough of Coq once to use this target [it will then
+correctly compute the deps and rebuild if you call the script again]
+This will be fixed in the future.
 
 For running in emacs, use `coqdev-ocamldebug` from `coqdev.el`.
 
+**Note**: If you are using OCaml >= 4.08 you need to use
```
Comment (SkySkimmer) [inline dev/doc/build-system.dune.md]: Can you write down why this is needed in this readme?

**Review by SkySkimmer — APPROVED** [review]

File: dev/doc/build-system.dune.md (hunk 1 above)
Comment (ejgallego) [reply dev/doc/build-system.dune.md]: Done.

File: dev/doc/build-system.dune.md [hunk 2]
```diff
@@ -108,24 +108,44 @@ automatically.
...
+or
+
+```
+(ocd) source dune_db_409
+```
+
+depending on your OCaml version. This is due to several factors:
+
+- OCaml >= 4.08 doesn't allow doubly-linking modules, however `source`
+  is not re entrant and seems to doubly-load in the default setup, see
```
Comment (SkySkimmer) [inline dev/doc/build-system.dune.md]: Should we report that upstream?

Comment (ejgallego) [reply dev/doc/build-system.dune.md]: Seems worthwhile, tho low priority. Note added in #8952

**Comment by ejgallego** [comment]
This broke the `lint` job on `master` builds but not on PRs, bizarre, must be some silly thing (cc @SkySkimmer )

**Comment by SkySkimmer** [comment]
It's failing the ocamlformat check `dune build @fmt`.
On PRs we do the per-commit builds so voboot was run.
I guess on master we should check that the current commit builds, then this would be fixed.

**Comment by ejgallego** [comment]
> I guess on master we should check that the current commit builds, then this would be fixed.

I see, that's a bit redundant tho; we could also just run `voboot` manually before the per-commit check.

Annoying `voboot` is going away real soon (TM)

**Comment by ejgallego** [comment]
#11438 proposes just to call the wrapper.

**Pull request merged**